src/tasks/ruuvi_task_adc.c

```diff
@@ -354,5 +354,51 @@ rd_status_t rt_adc_vdd_get (float * const battery)
 
     return err_code;
 }
+
+rd_status_t rt_adc_absolute_sample (rd_sensor_configuration_t * const configuration,
+                                    const uint8_t handle, float * const sample)
+{
+    rd_status_t err_code = RD_SUCCESS;
+    rd_sensor_data_t m_sample;
+    memset (&m_sample, RD_ADC_CLEAN_BYTE, sizeof (rd_sensor_data_t));
+    float battery_values;
+    m_sample.data = &battery_values;
+    m_sample.fields.datas.voltage_v = RD_ADC_DATA_COUNTER;
+    err_code |= rt_adc_init();
+    err_code |= rt_adc_configure_se (configuration, handle, ABSOLUTE);
+    m_vdd_prepared = (RD_SUCCESS == err_code);
+
+    if (RD_ERROR_INVALID_PARAM != err_code)
```

Comment:
Check against m_vdd_prepared? Now sampling would be tried on other errors than invalid param